br/pkg/storage/locking.go

```diff
@@ -3,23 +3,130 @@
 package storage
 
 import (
+	"bytes"
 	"context"
+	"encoding/hex"
 	"encoding/json"
 	"fmt"
+	"math/rand"
 	"os"
+	"path"
 	"time"
 
+	"github.com/google/uuid"
 	"github.com/pingcap/errors"
+	"github.com/pingcap/failpoint"
 	"github.com/pingcap/log"
 	"github.com/pingcap/tidb/br/pkg/logutil"
+	"go.uber.org/multierr"
 	"go.uber.org/zap"
 )
 
+// ExclusiveWrite is a write that in a strong consistency storage.
+//
+// It is pretty like a "write if not exist", but it is atomic:
```

Comment:
not sure this description is accurate cuz the `VerifyWriteContext ` can be arbitrary